src/main.ts

```diff
@@ -152,11 +155,19 @@ const createWindow = async () => {
   }
 
   mainWindow.once('ready-to-show', () => configureWatcher(mainWindow))
+  proxyEmitter.on('status:change', (statusName: ProxyStatus) => {
+    proxyStatus = statusName
+    mainWindow.webContents.send('proxy:status:change', statusName)
+  })
+  mainWindow.on('closed', () =>
+    proxyEmitter.removeAllListeners('status:change')
+  )
 
   return mainWindow
 }
 
 app.whenReady().then(async () => {
+  appSettings = await getSettings()
```

Comment:
Retrieving settings is the first event when the app launches. `getSettings` is an upsert so it'll create a configuration file if none is found.